MapboxNavigation/NavigationView.swift

```diff
@@ -38,26 +38,132 @@ import MapboxDirections
 @objc(MBNavigationView)
 open class NavigationView: UIView {
     
-    weak var mapView: NavigationMapView!
-    weak var wayNameLabel: WayNameLabel!
-    weak var bottomBannerView: BottomBannerView!
-    weak var bottomBannerContentView: BottomBannerContentView!
-    weak var instructionsBannerView: InstructionsBannerView!
-    weak var instructionsBannerContentView: InstructionsBannerContentView!
-    weak var lanesView: LanesView!
-    weak var nextBannerView: NextBannerView!
-    weak var statusView: StatusView!
-    weak var resumeButton: ResumeButton!
-    // Vertically laid-out stack view below the instructions banner consisting of StatusView, NextBannerView, and LanesView.
-    weak var informationStackView: UIStackView!
-    // Vertically laid-out stack view below the information stack view ontop of the map view, docked
-    // to the top right, consisting of Overview, Mute, and Report button.
-    weak var floatingStackView: UIStackView!
-    weak var overviewButton: FloatingButton!
-    weak var muteButton: FloatingButton!
-    weak var reportButton: FloatingButton!
-    weak var rerouteReportButton: ReportButton!
-    weak var separatorView: SeparatorView!
+    lazy var bannerShowConstraints: [NSLayoutConstraint] = [
+        self.instructionsBannerView.topAnchor.constraint(equalTo: self.safeTopAnchor),
+        self.instructionsBannerContentView.topAnchor.constraint(equalTo: self.topAnchor)]
+    
+    lazy var bannerHideConstraints: [NSLayoutConstraint] = [
+        self.informationStackView.bottomAnchor.constraint(equalTo: self.topAnchor),
+        self.instructionsBannerContentView.topAnchor.constraint(equalTo: self.instructionsBannerView.topAnchor)
+    ]
+    
+    lazy var endOfRouteShowConstraint: NSLayoutConstraint? = self.endOfRouteView?.bottomAnchor.constraint(equalTo: self.safeBottomAnchor)
+    
+    lazy var endOfRouteHideConstraint: NSLayoutConstraint? = self.endOfRouteView?.topAnchor.constraint(equalTo: self.bottomAnchor)
+    
+    lazy var endOfRouteHeightConstraint: NSLayoutConstraint? = self.endOfRouteView?.heightAnchor.constraint(equalToConstant: 260)
+    
+    lazy var rerouteFeedbackTopConstraint: NSLayoutConstraint = self.rerouteReportButton.topAnchor.constraint(equalTo: self.informationStackView.bottomAnchor, constant: 10)
+    
+    private struct Images {
+        static let overview = UIImage(named: "overview", in: .mapboxNavigation, compatibleWith: nil)!.withRenderingMode(.alwaysTemplate)
+        static let volumeUp = UIImage(named: "volume_up", in: .mapboxNavigation, compatibleWith: nil)!.withRenderingMode(.alwaysTemplate)
+        static let volumeOff =  UIImage(named: "volume_off", in: .mapboxNavigation, compatibleWith: nil)!.withRenderingMode(.alwaysTemplate)
+        static let feedback = UIImage(named: "feedback", in: .mapboxNavigation, compatibleWith: nil)!.withRenderingMode(.alwaysTemplate)
+    }
```

Comment:
Maybe we shouldn't force unwrap the images here and allow whoever uses the Images struct to unwrap them. It's safer that way.

Reply:
![Optionals!](https://i.imgur.com/k0S28Du.gif)

Optionals are considered desirable when, _in the course of normal operation_, a developer could encounter a case where the property/variable/pony might not exist. 

In this case, since the images are bundled in the framework, the developer should not normally expect a situation where those images are `nil`. In this sense, we use the force-unwrap operator like an assertion -- we ensure that something that _should_ be there (out of the developer's control), and then free up the developer from having to make that assurance upon every invocation. 

Consider the following -- if our image asset was missing, would it be better for the developer to receive an assertion, a recoverable error, or nothing at all? Often times, it's desirable to "fail fast" when "we" make a mistake with _our_ framework, as wrapping such unlikely scenarios in optionals can lead to subtle, very-hard-to-diagnose bugs down the line.

However, this struct should really be a careless enum -- something I forgot to do whilst working on this. Thanks for reminding me!

/cc @1ec5 @akitchen

Reply:
I think in this case, force unwrapping is fine.